### PR TITLE
Create Site and Location CRUD

### PR DESCRIPTION
**Pull request by brec9824**
Create routes and controllers for site and location CRUD. The location.show page includes a paginated list of its connected devices. The site.show page includes a paginated list of its connected locations. Add feature #83

### CODE REVIEW

**Comment by brandonjank** [comment]
I'll review this tonight after work.

File: resources/views/location/edit.blade.php [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+					<div class="panel-body">
+						{!! Form::model($location, array('route' => ['location.update', $location->id], 'method' => 'PUT', 'class' => 'form-horizontal')) !!}
+						@include('location.form')
+						<div class="form-group">
+							<div class="col-md-6 col-md-offset-4">
+								{!! Form::button('<i class="glyphicon glyphicon-ok"></i> Save', array('type' => 'submit', 'class' => 'btn btn-info pull-right')) !!}
+								<a href="{{ route('location.show', $location->id) }}" class="btn btn-primary pull-right"><i class="glyphicon glyphicon-remove"></i> Cancel</a>
+							</div>
+						</div>
+						{!! Form::close() !!}
```
Comment (brandonjank) [inline resources/views/location/edit.blade.php]: {!! Form::close() !!} should be in the same div as the button, line 18.

File: resources/views/location/create.blade.php [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+					<div class="panel-body">
+						{!! Form::open(array('route' => 'location.store', 'method' => 'POST', 'class' => 'form-horizontal')) !!}
+						@include('location.form')
+						<div class="form-group">
+							<div class="col-md-6 col-md-offset-4">
+								{!! Form::button('<i class="glyphicon glyphicon-ok"></i> Create', ['type' => 'submit', 'class' => 'btn btn-info pull-right']) !!}
+								<a href="{{ route('location.index') }}" class="btn btn-primary pull-right"><i class="glyphicon glyphicon-remove"></i> Cancel</a>
+							</div>
+						</div>
+						{!! Form::close() !!}
```
Comment (brandonjank) [inline resources/views/location/create.blade.php]: {!! Form::close() !!} should be in the same div as the button, line 18.

File: routes/web.php [hunk 3]
```diff
@@ -89,3 +89,13 @@
  * Image Controller  *
  *-----------------------*/
 Route::get('image/device/{device_id}', 'ImageController@show');
+
+/*-----------------------*
+ * Location Controller *
+ *-----------------------*/
+Route::resource('location', 'LocationController');
```
Comment (brandonjank) [inline routes/web.php]: Add location.index and site.index to manage menu in layouts/dashboard and layouts/home as Manage Locations and Manage Sites

Comment (brandonjank) [reply routes/web.php]: Create a ON DELETE CASCADE foreign key relationship between sites and locations. So when a site is deleted so are it's locations. Something like this in a new migration:
```php
		Schema::table('locations', function(Blueprint $table) {
			$table->foreign('site_id')->references('id')->on('sites')
						->onDelete('cascade');
		});
```

Comment (brec9824) [reply routes/web.php]: @brandonjank Added location.index and site.index to the home layout manage drop down list. They were already added to the dashboard layout. Also the delete cascade for when a site is deleted its connected locations also delete is already implemented.

**Pull request merged**